libs/go/zmscli/domain.go

```diff
@@ -38,7 +60,7 @@ func (cli Zms) DeleteDomain(dn string) (*string, error) {
 		}
 		if err == nil {
 			s := "[Deleted domain " + dn + "]"
-			return &s, nil
+			return cli.switchOverFormats(&StandardJSONMessage{Message: s}, s)
```

Comment:
I noticed that you wrap the output with StandardJSONMessage before calling switchOverFormats.
The assumption here is that there are only 2 formats:
1. The default (yaml) - which will use the second argument not wrapped
2. Json - which will this one.
This will work fine for 2 formats but if we would like to add a third in the future, say XML, this will need to change.
Please try to right it in such a way that adding new formats will not require changing the existing calls.

Reply:
Makes sense, let me change that function definition up a bit.

Reply:
@OferLevi85, take a look at https://github.com/AthenZ/athenz/pull/1386/commits/84490b54393e1d9f4033ee326e622059849498d3#diff-8118320d935b49c9f953eebaf5401929b9737f08b783fe1a160e4421109acbcfL30-L38 and let me know what you think.

Reply:
With these new changes, any new output format can be supported without having to change the function call statements for single-line messages or for native Go structs in the future.

Reply:
Thanks I'll take a look.
By the way, please implement the flag to all methods, not just domain methods.

Reply:
Could you create a separate issue maybe for that? There are quite a few commands to cover.